inst/shiny/modules/tab_visuals.R

```diff
@@ -0,0 +1,560 @@
+    # This tab computes and visualizes output data from the NCA analysis
+
+    # picker Input to filter for parameters to display in the summary table
+    output$summaryselect <- renderUI({
+      req(res_nca())
+
+      # available parameters
+      paramselection <- unique(res_nca()$result$PPTESTCD)
+      # select from available with all preselected
+      pickerInput(
+        ns("paramselect"),
+        "Filter parameters to display:",
+        choices = paramselection,
+        selected = paramselection,
+        multiple = TRUE,
+        options = list(`actions-box` = TRUE)
+      )
+    })
+
+    # Update inputs based on what is available in the data
+    observeEvent(res_nca(), {
+      # Define the relevant columns for the group by picker
+      group_cols <- unname(unlist(res_nca()$data$conc$columns$groups))
+      classification_cols <- sort(
+                                  c("SEX", "RACE", "ACTARM", "AGE",
+                                    "TRT01P", "TRT01A", "DOSEA"))
+      classification_cols <- classification_cols[
+        classification_cols %in% names(res_nca()$data$conc$data)
+      ]
+
+      # update the input for the group by picker
+      updateOrderInput(session, "summarygroupbysource",
+                       items = c(group_cols, classification_cols))
+    })
+
+    # Reactive expression for summary table based on selected group and parameters
+    summary_stats <- reactive({
+      req(input$summarygroupby, input$paramselect)
+
+      # Calculate summary stats and filter by selected parameters
+      calculate_summary_stats(res_nca(), input$summarygroupby) %>%
+        filter(PPTESTCD %in% input$paramselect)  %>%
+        rename(PARAM = PPTESTCD)
+    })
+
+    # render the reactive summary table in a data table
+    output$descriptivestats <- DT::renderDataTable({
+      req(summary_stats())
+      DT::datatable(
+        data = summary_stats(),
+        extensions = "FixedHeader",
+        options = list(
+          scrollX = TRUE,
+          scrollY = TRUE,
+          lengthMenu = list(c(10, 25, -1), c("10", "25", "All")),
+          fixedHeader = TRUE
+        )
+      )
+    })
+
+    output$descriptivestats2 <- DT::renderDataTable({
+      req(res_nca())
+      DT::datatable(
+        data = calculate_summary_stats(res_nca())  %>% rename(PARAM = PPTESTCD),
+        extensions = "FixedHeader",
+        options = list(
+          scrollX = TRUE,
+          scrollY = TRUE,
+          lengthMenu = list(c(10, 25, -1), c("10", "25", "All")),
+          fixedHeader = TRUE
+        )
+      )
+    })
+
+    # Save summary stats to improve
+    observeEvent(input$downloadsum, {
+      showModal(modalDialog(
+        title = "Please enter the path to the folder on improve for your results:",
+        textInput(ns("pathresults"), "Path:"),
+        textInput(ns("filename"), "Enter filename of choice:", ""),
+        actionButton(ns("go2"), "GO"),
```

Comment:
issue: This button does not seem to have any event attached.
issue: If this is functional button, it needs more descriptive name.,